spec/index.html

```diff
@@ -10187,8 +10191,39 @@ <h3>SPARQL Algebra</h3>
         <p>The <a href="#defn_algReduced" class="algFct">Reduced</a> solution sequence modifier does not guarantee a defined multiplicity.</p>
         <div class="defn">
           <p><b>Definition: <span id="defn_algSlice">Slice</span></b></p>
-          <p>Let <var>Ψ</var> be a sequence of solution mappings. We define:</p>
-          <p><a href="#defn_algSlice" class="algFct">Slice</a>(<var>Ψ</var>, <var>start</var>, <var>length</var>)[<var>i</var>] = <var>Ψ</var>[<var>start</var>+<var>i</var>] for <var>i</var> = 0 to (<var>length</var>-1)</p>
+          <p>Let <var>Ψ</var> be a sequence of solution mappings, and |offset| and |length| be integers. We define:</p>
```

Comment:
Neither offset or limit can be negative by the grammar.

[15.5 LIMIT](https://www.w3.org/TR/sparql12-query/#modResultLimit)  "A limit may not be negative."

```
[28]      LimitClause    ::=  'LIMIT' INTEGER
[29]      OffsetClause   ::=  'OFFSET' INTEGER
[166]     INTEGER        ::=  [0-9]+
```

Given negative lengths in some programming languages mean "from the end"

(sorry - it can be confusing - but that's just the way the grammar is to deal with unary operators)

```suggestion
          <p>Let <var>Ψ</var> be a sequence of solution mappings, and |offset| and |length| be non-negative integers. We define:</p>
```

then simplify the definitions.